server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestRolloverIndexAction.java

```diff
@@ -46,6 +47,11 @@ public String getName() {
         return "rollover_index_action";
     }
 
+    @Override
+    public Set<String> supportedCapabilities() {
+        return Set.of("lazy-rollover-failure-store");
```

Comment:
So, I think we should only expose this conditionally as part of the feature flag being on, otherwise we add a capability to the general release that is not actually supported yet.

```suggestion
        if (DataStream.isFailureStoreFeatureFlagEnabled()) {
            return Set.of("lazy-rollover-failure-store");
        } else {
            return Set.of();
        }
```

I'd appreciate someone poking holes in this idea though. I'm engaging with the core infra folks on how best to approach using these new API's with feature flags. Unfortunately, we've mostly spun our wheels on coming up with a best practice. Right now, I think conditionally registering the capability to the rest end point is probably fine to start, but when we remove the feature flag we'll need to go through and make sure features/capabilities are correctly tagged up.